### DISCUSSION TITLE
gdb-dashboard installation

### DISCUSSION THREAD

**Discussion by ColinBathe** [stnolting/neorv32#393] [Show and tell]
gdb-dashboard is a useful addition to gdb. This is what you need to get it operational.

When compiling the toolchain you need to include --with-python as an option.
```
sudo ./configure --with-python --prefix=/opt/riscv --with-arch=rv32gc --with-abi=ilp32
```
This will give you a warning that --with-python is an unrecognised option but this is only at the top level. The option ripples down to the gdb directory where it is needed.

Installation of gdb-dashboard then follows the instructions here: 
[https://github.com/cyrus-and/gdb-dashboard](https://github.com/cyrus-and/gdb-dashboard)
```
cd ~
wget -P ~ https://git.io/.gdbinit
sudo apt install python3-pip
pip install pygments
```

Run gdb as normal and you get the additional display.
![image](https://user-images.githubusercontent.com/24269597/186170361-45101fcd-0e3a-4dd2-99ea-b1c2bf92f579.png)

**Comment by stnolting** [comment]
Looks really good!

Btw, I just updated my pre-built RISC-V gcc toolchains in https://github.com/stnolting/riscv-gcc-prebuilt but without the `--with-python` switch. I copied the `.gdbinit` file into my home folder and I just found out that my GDB now looks pretty as well 😄

![gdb](https://user-images.githubusercontent.com/22944758/186183490-cf4d5b4e-6948-431e-9794-b549e8f223a2.png)

So maybe `--with-python` is set by default (somewhere under the hood)?! 🤔 Anyway, thanks for the hint!

**Reply by ColinBathe** [reply]
It didn't work for me without the --with-python when I tried it. gdb complained that it didn't have python integrated and a recompile with the flag was needed to get it to work. Possibly we don't have the same setup though.

**Reply by DS-567** [reply]
This works well actually. How do you stop this opening automatically upon running GDB?

Is there a way to choose if the GDB dashboard opens or not when running GDB?

**Reply by stnolting** [reply]
Maybe there is a switch for it, I am not sure. But if you put that `.gdbinit` into your (Ubuntu) home folder it is _always_ automatically loaded when starting GDB.

**Reply by DS-567** [reply]
Thanks Stephan!
